### PR TITLE
Add Workers docs article

### PR DESCRIPTION
**Pull request by YannBCF**
Add workers documentation article for the clientside Workers API implemented in dev7.0-dev9

### CODE REVIEW

File: docs/articles/workers.md [hunk 1]
```diff
@@ -92,4 +105,76 @@ client.js
...
     console.log(args);
 });
+```
+
+### Using shared array buffers
+
+An [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) is an object used to represent a generic, fixed-length raw binary data buffer.
+It is an array of bytes, you cannot directly manipulate its content but create a [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) object / [DataView](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) to read and write the buffer content.
+
+A [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) is very similar to an array buffer except he is built in a way that they can be used to create views on shared memory, unlike an ArrayBuffer it cannot become detached.
```
Comment (LeonMrBonnie) [inline docs/articles/workers.md:115]: `except he is built`
A shared array buffer does not have a gender

File: docs/articles/workers.md [hunk 2]
```diff
@@ -92,4 +105,76 @@ client.js
...
+
+**example usage**
```
Comment (LeonMrBonnie) [inline docs/articles/workers.md:117]: Wrong casing

File: docs/articles/toc.yml [hunk 3]
```diff
@@ -1,3 +1,5 @@
+- name: Workers
```
Comment (C0kkie) [inline docs/articles/toc.yml:1]: Move workers to the bottom

File: docs/articles/workers.md [hunk 4]
```diff
@@ -92,4 +105,76 @@ client.js
...
+        const oldValue = Atomics.exchange(sharedArray, i, 5)
+        console.log(`[Worker] Value change to 5 on [${i}]`);
+
+        // reading the new value
+        const changedValue = Atomics.load(sharedArray, i);
+        console.log(`[Worker] New value on ${i} is ${changedValue} (old value: ${oldValue})`);
+    }
+
+    // You can put this inside the for-loop and it will print out all console.logs that are triggered by this event once the atomics operations are finished.
+    // I put it outside of my for-loop because my console got spammed xd
```
Comment (LeonMrBonnie) [inline docs/articles/workers.md:177]: This comment does not belong in docs

File: docs/articles/workers.md [hunk 5]
```diff
@@ -38,6 +35,22 @@ if you do ``setInterval`` or ``console.log`` it will call the corresponding alt
 
 ### How to create a worker
 
+# [main.js](#tab/tab1-0)
```
Comment (LeonMrBonnie) [inline docs/articles/workers.md:38]: What are the import assertions doing here?

File: docs/articles/workers.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+It is an array of bytes, you cannot directly manipulate its content but create a [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) object / [DataView](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) to read and write the buffer content.
+
+A [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) is very similar to an array buffer except he is built in a way that they can be used to create views on shared memory, unlike an ArrayBuffer it cannot become detached.
+
+**example usage**
+
+> [!WARNING]
+> You should learn how to use SharedArrayBuffers outside of the alt:V context in order to use them properly. The following example is making use of the [Atomics](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Atomics) API to make the SharedArrayBuffer usage [thread safe](https://en.wikipedia.org/wiki/Thread_safety)
+
+client.js
```
Comment (C0kkie) [inline docs/articles/workers.md:122]: I would prefer tabs here for client.js and worker.js

File: docs/articles/workers.md [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+worker.js
+```js
+alt.on('eventName', (...args) => {
+    console.log(args);
+});
+```
+
+### Communicating from the worker to the client
+
+worker.js
```
Comment (C0kkie) [inline docs/articles/workers.md:98]: I would prefer tabs here for client.js and worker.js

File: docs/articles/workers.md [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+
+```js
+worker.on('error', (error) => {
+    console.log(error);
+});
+```
+
+### Communicating from the client to the Worker
+
+client.js
```
Comment (C0kkie) [inline docs/articles/workers.md:84]: I would prefer tabs here for client.js and worker.js

File: docs/articles/workers.md [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+worker.start();
+```
+
+Using absolute path (starting from the resource root directory).
+```js
+const worker = new alt.Worker('/client/worker.js');
+worker.start();
+```
+
+### Knowing when the worker is loaded
```
Comment (LeonMrBonnie) [inline docs/articles/workers.md:66]: You should probably have a bigger header called like `Built-in worker events` or something similiar

**Review by C0kkie — CHANGES_REQUESTED** [review]

File: docs/articles/workers.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+    console.log(args);
+});
+```
+
+### Using shared array buffers
+
+An [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) is an object used to represent a generic, fixed-length raw binary data buffer.
+It is an array of bytes, you cannot directly manipulate its content but create a [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) object / [DataView](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) to read and write the buffer content.
+
+A [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) is very similar to an array buffer except he is built in a way that they can be used to create views on shared memory, unlike an ArrayBuffer it cannot become detached.
```
Comment (LeonMrBonnie) [inline docs/articles/workers.md:115]: Remove the last part about detaching an array buffer, this is not something the average developer cares about. (And is actually not even possible in native JS, only through the V8 C++ API)

File: docs/articles/workers.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+### Using shared array buffers
+
+An [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) is an object used to represent a generic, fixed-length raw binary data buffer.
+It is an array of bytes, you cannot directly manipulate its content but create a [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) object / [DataView](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView) to read and write the buffer content.
+
+A [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer) is very similar to an array buffer except it is built in a way that they can be used to create views on shared memory.
+
+**Example usage**
+
+> [!WARNING]
```
Comment (C0kkie) [inline docs/articles/workers.md:106]: I think putting the warning above ``**Example usage**`` would make more sense

**Review by C0kkie — APPROVED** [review]

**Review by LeonMrBonnie — APPROVED** [review]

**Pull request merged**